ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -748,15 +748,29 @@ private[kvutils] class TransactionCommitter(
 }
 
 private[kvutils] object TransactionCommitter {
-  case class DamlTransactionEntrySummary(submission: DamlTransactionEntry) {
+  class DamlTransactionEntrySummary(val submission: DamlTransactionEntry, tx: => Tx.Transaction) {
     val ledgerEffectiveTime: Timestamp = parseTimestamp(submission.getLedgerEffectiveTime)
     val submitterInfo: DamlSubmitterInfo = submission.getSubmitterInfo
     val commandId: String = submitterInfo.getCommandId
     val submitters: List[Party] =
       submitterInfo.getSubmittersList.asScala.toList.map(Party.assertFromString)
-    lazy val transaction: Tx.Transaction = Conversions.decodeTransaction(submission.getTransaction)
+    lazy val transaction: Tx.Transaction = tx
     val submissionTime: Timestamp = Conversions.parseTimestamp(submission.getSubmissionTime)
     val submissionSeed: crypto.Hash = Conversions.parseHash(submission.getSubmissionSeed)
+    // On copy, avoid decoding the transaction again if not needed
```

Comment:
Please add a newline here.

```suggestion

    // On copy, avoid decoding the transaction again if not needed
```